### PR TITLE
Support of mixed version upgrade and rdo type jobs with TrasS.

### PR DESCRIPTION
**Pull request by sathlan**
Those patches enable TraaS to run mixed version upgrade jobs and add missing bit for rdo type job.  Some convenient toolings are added and swap partition are automatically created to be able to work with medium instances on rdo cloud.

### CODE REVIEW

File: scripts/traas.sh [hunk 1]
```diff
@@ -15,6 +15,8 @@ export PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 export TOCI_JOBTYPE=${TOCI_JOBTYPE:-"multinode-nonha-oooq"}
 export MTU=1400
 export DO_SETUP_NODEPOOL_FILES=${DO_SETUP_NODEPOOL_FILES:-"1"}
+export NODEPOOL_REGION=${NODEPOOL_REGION:-dfw}
+export NODEPOOL_CLOUD=${NODEPOOL_CLOUD:-rax}
```
Comment (slagle) [inline scripts/traas.sh:19]: Can you default these to more generic values? just "traas" for both perhaps?

File: scripts/traas.sh [hunk 2]
```diff
@@ -57,52 +57,53 @@ source workspace/.quickstart/bin/activate
...
     if [ "$project" = "openstack/tripleo-quickstart-extras" ]; then
-        rm -rf tripleo-quickstart-extras
-        git clone https://git.openstack.org/openstack/tripleo-quickstart-extras
-        pushd tripleo-quickstart-extras
-        git fetch https://git.openstack.org/openstack/tripleo-quickstart-extras $ref && git checkout FETCH_HEAD
-        pip install .
-        popd
+        curl https://review.openstack.org/changes/${change}/revisions/current/patch?download | \
+            base64 -d | \
+            sudo patch -f -d ./tripleo-quickstart-extras -p1
```
Comment (slagle) [inline scripts/traas.sh:97]: you don't need the if/else logic at all anymore do you? just use $project in your patch command.

File: scripts/traas.sh [hunk 3]
```diff
@@ -46,7 +46,7 @@ check_var TOCI_JOBTYPE
 # Update openssh before the test builds are installed by tripleo-ci
 # See https://review.openstack.org/#/c/437683/
 sudo yum -y update openssh
-sudo yum -y install patch
+sudo yum -y install patch wget
```
Comment (slagle) [inline scripts/traas.sh]: we can have this here, but it should absolutely also be in quickstart if wget is still a requirement that doesn't get installed automatically.

**Comment by bogdando** [comment]
testing this locally, it's ended up with `overcloud-0 | ERROR  | None       | NOSTATE `. Not sure if related to the patch, will have to test more.

**Comment by sathlan** [comment]
@bogdando this looks more like a cloud issue to me.  But to tell the truth I didn't yet test that branch, so maybe there is an issue there.  Will setup a test of that particular branch.  Thank for the feedback.

File: templates/traas.yaml [hunk 4]
```diff
@@ -156,7 +156,7 @@ resources:
         export HOME=/root
         BRANCH=${BRANCH#stable/}
         env STABLE_RELEASE=${BRANCH} tripleo-ci/scripts/tripleo.sh --repo-setup
-        yum -y install python-heat-agent*
+        yum -y install python-heat-agent* jq vim
```
Comment (slagle) [inline templates/traas.yaml]: are these dependencies for the job itself? or just to make it useful while interactively debugging? if it's for the job itself, it should be in tripleo-quickstart itself.

**Review by slagle — COMMENTED** [review]
looks fine to merge, left a few comments

File: scripts/traas.sh (hunk 2 above)
Comment (sathlan) [reply scripts/traas.sh:97]: yeah, right.  Will do.  Thanks.

File: templates/traas.yaml (hunk 4 above)
Comment (sathlan) [reply templates/traas.yaml]: Just debugging.  jq is usually installed on overcloud like image but may be missing on other image.  vim, it's just that it's annoying to not have it.

File: scripts/traas.sh (hunk 1 above)
Comment (sathlan) [reply scripts/traas.sh:19]: Hum, I don't remember their purpose.  I will try with non relevant value.

**Comment by sathlan** [comment]
Thanks for the merge.  More is coming :)

**Pull request merged**